UPGRADING.md

```diff
@@ -0,0 +1,105 @@
+# Upgrading
+
+This section describes how to upgrade a Conjur Server.
+
+## Standard Process
+
+The following steps describe a standard upgrade of the Conjur server, when deployed
+using Docker Compose. These steps assume you have defined your Conjur image in
+a service named `conjur`, and that you have access to the Conjur data key
+that was used when you originally deployed your Conjur server.
+1. In your terminal window, set the `CONJUR_DATA_KEY` environment variable:
+   ```
+   export CONJUR_DATA_KEY={your Conjur data key}
+   ```
+
+1. Edit the Conjur image version in `docker-compose.yml` to reference the new
+   version.
+
+1. Pull the new Conjur image version:
+   ```
+   docker-compose pull conjur
+   ```
+
+1. Stop the Conjur container:
+   ```
+   docker-compose stop conjur
+   ```
+
+1. Bring up the Conjur service using the new image version without changing
+   linked services:
+   ```
+   docker-compose up -d --no-deps conjur
+   ```
+
+1. View Docker containers and verify all are healthy, up and running:
+   ```
+   docker ps -a
+   ```
+
+   It may also be useful to check the logs of the Conjur
+   container to ensure that Puma started successfully, which can be done by
+   running
+   ```
+   docker logs conjur_server
+   ```
+   where `conjur_server` is defined as the `container_name` of the Conjur
+   service in the `docker-compose.yml`.
+   
+
+### Troubleshooting
+
+If you run through the steps above _without_ setting the `CONJUR_DATA_KEY`
+environment variable first, you will be able to complete the steps successfully
+but the logs of the new Conjur container will show an error like:
+```
+$ docker logs conjur_server
+rake aborted!
+No CONJUR_DATA_KEY
+...
+```
+`conjur_server` in the example command above is defined as the `container_name`
+of the `conjur` service in the `docker-compose.yml`.
+
+To fix this, set the `CONJUR_DATA_KEY` environment variable and run through
+the process again. This time when you check the logs of the Conjur server
+container you should see the Puma service starting as expected:
+```
+$ docker logs conjur_server
```

Comment:
Maybe this command should be better? encapsulate with CLI
`docker-compose exec conjur conjurctl wait`

![image](https://user-images.githubusercontent.com/50704381/87647657-229d3900-c758-11ea-9622-7f13d28df924.png)

Reply:
I'm glad you suggested that! I use that command myself and was trying to find a way to work it in - I hesitated, because I'm not sure it works well enough when the server fails. I'll play around with it and see how it goes.